qubesmanager/utils.py

```diff
@@ -135,3 +135,31 @@ def debug(*args, **kwargs):
     if not is_debug():
         return
     print(*args, **kwargs)
+
+
+def get_path_from_vm(vm, service_name):
+    """
+    Displays a file/directory selection window for the given VM.
+
+    :param vm: vm from which to select path
+    :param service_name: qubes.SelectFile or qubes.SelectDirectory
+    :return: path to file, checked for validity
+    """
+
+    path_re = re.compile(r"[a-zA-Z0-9/:.,_+=() -]*")
+    path_max_len = 512
+
+    if not vm:
+        return None
+    stdout, stderr = vm.run_service_for_stdio(service_name)
+
+    untrusted_path = stdout.decode()[:path_max_len]
```

Comment:
Better use `ascii` decoder (argument to `decode()`). Non-ASCII characters are filtered out by `path_re` anyway.